data/bash-scripts/resizing_images_in_folder.bash

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+# Script to resize all images in a directory to 640x400 pixels
+# Requires ImageMagick to be installed
+
+# Default directory is current directory
+DIR="${1:-.}"
+
+# Check if directory exists
+if [ ! -d "$DIR" ]; then
+    echo "Error: Directory '$DIR' does not exist."
+    exit 1
+fi
+
+# Create output directory for resized images
+OUTPUT_DIR="$DIR/resized"
+mkdir -p "$OUTPUT_DIR"
+
+# Supported image extensions
+extensions=("jpg" "jpeg" "png")
+
+echo "Resizing images in directory: $DIR"
+echo "Output directory: $OUTPUT_DIR"
+echo "Target size: 576x320"
+echo "####################"
+
+# Process each image type
+for ext in "${extensions[@]}"; do
+    # Process both lowercase and uppercase extensions
+    for file in "$DIR"/*."$ext" "$DIR"/*."${ext^^}"; do
+        if [ -f "$file" ]; then
+            filename=$(basename "$file")
+            echo "Processing: $filename"
+            
+            # Resize image maintaining aspect ratio, then crop/pad to exact dimensions
+            convert "$file" -resize 576x320^ -gravity center -extent 576x320 "$OUTPUT_DIR/$filename"
```

Comment:
Maybe you can add variables for height and width but leave to you to add them or not?